### PR TITLE
ENG-388 Implement handling of DE component usage on uninstall

### PR DESCRIPTION
**Pull request by raxkaynan**

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **nicpuddu**

If they are not already assigned, you can assign the PR to them by writing `/assign @nicpuddu` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by entando-jx** [comment]
@raxkaynan: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 70dae156010a2121f0872ec93010ded61234f598 | [link](https://raw.githubusercontent.com/entando/devops-results/logs/jenkins-x/logs/entando/app-builder/PR-692/1.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@raxkaynan: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 37f81ac64d95247dfbe48614d179d42601549226 | [link](https://raw.githubusercontent.com/entando/devops-results/logs/jenkins-x/logs/entando/app-builder/PR-692/2.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: src/ui/digital-exchange/components/common/ComponentInstallActions.js [hunk 1]
```diff
@@ -124,6 +129,7 @@ ComponentInstallActions.propTypes = {
   onRecheckStatus: PropTypes.func.isRequired,
   onRetryAction: PropTypes.func.isRequired,
   installUninstallLoading: PropTypes.bool.isRequired,
+  componentUsageList: PropTypes.arrayOf(PropTypes.object).isRequired,
```
Comment (jeffgo10) [inline src/ui/digital-exchange/components/common/ComponentInstallActions.js:132]: pls avoid `PropTypes.object`. Try to use `PropTypes.shape` instead (even better with detailed shape of object)

File: src/ui/digital-exchange/components/common/ConfirmUninstallModal.js [hunk 2]
```diff
@@ -44,6 +57,7 @@ ConfirmUninstallModal.propTypes = {
   info: PropTypes.shape({
     id: PropTypes.string.isRequired,
     name: PropTypes.string,
+    usageList: PropTypes.arrayOf(PropTypes.object),
```
Comment (jeffgo10) [inline src/ui/digital-exchange/components/common/ConfirmUninstallModal.js:60]: here too regarding `PropTypes.object`

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/692#pullrequestreview-398129655" title="Approved">jeffgo10</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **nicpuddu**

If they are not already assigned, you can assign the PR to them by writing `/assign @nicpuddu` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**